x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java

```diff
@@ -291,6 +292,70 @@ public void testSnapshotUserRole() {
         assertNoAccessAllowed(snapshotUserRole, RestrictedIndicesNames.ASYNC_SEARCH_PREFIX + randomAlphaOfLengthBetween(0, 2));
     }
 
+    public void testEnrichUserRole() {
+        final TransportRequest request = mock(TransportRequest.class);
+        final Authentication authentication = mock(Authentication.class);
+
+        RoleDescriptor roleDescriptor = new ReservedRolesStore().roleDescriptor("enrich_user");
+        assertNotNull(roleDescriptor);
+        assertThat(roleDescriptor.getMetadata(), hasEntry("_reserved", true));
+
+        Role enrichUserRole = Role.builder(roleDescriptor, null).build();
+        assertThat(enrichUserRole.cluster().check(PutPipelineAction.NAME, request, authentication), is(true));
+        assertThat(enrichUserRole.cluster().check(GetPipelineAction.NAME, request, authentication), is(true));
+        assertThat(enrichUserRole.cluster().check(DeletePipelineAction.NAME, request, authentication), is(true));
+        assertThat(enrichUserRole.cluster().check(PutIndexTemplateAction.NAME, request, authentication), is(false));
+        assertThat(enrichUserRole.cluster().check(GetIndexTemplatesAction.NAME, request, authentication), is(false));
+        assertThat(enrichUserRole.cluster().check(DeleteIndexTemplateAction.NAME, request, authentication), is(false));
+        assertThat(enrichUserRole.cluster().check(ClusterRerouteAction.NAME, request, authentication), is(false));
+        assertThat(enrichUserRole.cluster().check(ClusterUpdateSettingsAction.NAME, request, authentication), is(false));
+        assertThat(enrichUserRole.cluster().check(MonitoringBulkAction.NAME, request, authentication), is(false));
+        assertThat(enrichUserRole.cluster().check(DelegatePkiAuthenticationAction.NAME, request, authentication), is(false));
+        assertThat(enrichUserRole.cluster().check(MainAction.NAME, request, authentication), is(true));
+        assertThat(enrichUserRole.cluster().check(XPackInfoAction.NAME, request, authentication), is(true));
+        assertThat(enrichUserRole.cluster().check(RemoteInfoAction.NAME, request, authentication), is(true));
+        // has monitor
+        assertThat(enrichUserRole.cluster().check(ClusterHealthAction.NAME, request, authentication), is(true));
+        assertThat(enrichUserRole.cluster().check(ClusterStateAction.NAME, request, authentication), is(true));
+        assertThat(enrichUserRole.cluster().check(ClusterStatsAction.NAME, request, authentication), is(true));
+        assertThat(enrichUserRole.cluster().check(PutIndexTemplateAction.NAME, request, authentication), is(false));
+        assertThat(enrichUserRole.cluster().check(ClusterRerouteAction.NAME, request, authentication), is(false));
+
+        assertThat(enrichUserRole.runAs().check(randomAlphaOfLengthBetween(1, 12)), is(false));
+
+        final String index = "enrich-." + randomAlphaOfLength(randomIntBetween(0, 13));
```

Comment:
Shouldn't this be
```suggestion
        final String index = ".enrich-" + randomAlphaOfLength(randomIntBetween(0, 13));
```
since the existing role grants access to index pattern of `.enrich-*`? But the PR's description says `enrich*` and the IT test uses `.enrich*`. Do we want to be consistent?

Reply:
> Do we want to be consistent?

Always.
This was obviously a typo. In English I use "enrich indices" whilst in code I use ".enrich-*" .
Sometimes mutants are spawned but which must not be tolerated!

Thanks for catching it.

Reply:
Thanks, do you mind fix the index pattern used in the description as well? Also the tests use both `.enrich*` and `.enrich-*` (the title uses `.enrich*`). Or maybe it does not matter at this level? Sorry for being paranoid.